include/tvm/runtime/packed_func.h

```diff
@@ -1996,8 +2123,10 @@ inline TObjectRef TVMPODValue_::AsObjectRef() const {
         << "Expect a not null value of " << ContainerType::_type_key;
     return TObjectRef(ObjectPtr<Object>(nullptr));
   }
-  // NOTE: the following code can be optimized by constant folding.
-  if (std::is_base_of<NDArray::ContainerType, ContainerType>::value) {
+
+  // NOTE: The following code uses "if constexpr" wherever possible to
+  // minimize the number of runtime checks.
```

Comment:
Sometimes I love modern C++ :slightly_smiling_face:

Reply:
Holy cow, yeah.  Each step from  C++03 partial template specialization, to C++11 `std::enable_if`, to C++17 `if constexpr` brings such an improvement in readability.  It probably results in the same optimized binary at the end, since the optimizer would (probably) DCE the constants, but this way the compiler will yell at me if I accidentally insert a runtime value into the condition.